master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Enable default deny for Kubernetes pods
+---
+
+### Big picture
+
+Change the default for Kubernetes pods from allowing all traffic, to denying all traffic using Kubernetes or Calico network policy.  
+
+### Value
+
+A **default deny** network policy provides an enhanced security posture -- so workloads without policy (or incorrect policy) are not allowed traffic until appropriate network policy is defined.
+
+### Features
+
+This how-to guide uses the following Calico features:
+- **NetworkPolicy** 
+- **GlobalNetworkPolicy**
+
+### Concepts
+
+#### Default deny/allow behavior
+
+**Default allow** means all traffic is allowed by default, unless otherwise specified. **Default deny** means all traffic is denied by default, unless explicitly allowed. **Kubernetes pods are default allow**, unless network policy is defined to specify otherwise.
+
+For compatibility with Kubernetes, **Calico network policy** enforcement follows the standard convention for Kubernetes pods:
+- If no network policies apply to a pod, then all traffic to/from that pod is allowed.
+- If one or more network policies apply to a pod with type ingress, then only the ingress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+- If one or more network policies apply to a pod with type egress, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**. Only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
+
+#### Best practice: implicit default deny policy
+
+Whether you use a network policy or global network policy, we recommend that you create a "plain vanilla" default deny policy that does not include other deny rules. This best practice ensures that unwanted traffic is denied by default. Note that implicit default deny policy always occurs last; if any other policy allows the traffic, then the deny does not come into effect. The deny is executed only after all other policies are evaluated. 
+
+### How to
+
+**Best practice**
```

Comment:
Should this be combined with the "Best practice" section above?

Reply:
It could, but I figured some people might skip Concepts and go right to How to. I wanted to be sure they parsed the options and could see why we put Calico global policy first in the list.

Reply:
sounds good to me